evm/src/fuzz/invariant/executor.rs

```diff
@@ -551,17 +554,28 @@ fn collect_data(
 
 /// Verifies that the invariant run execution can continue.
 /// Returns the mapping of (Invariant Function Name -> Call Result) if invariants were asserted.
+#[allow(clippy::too_many_arguments)]
 fn can_continue(
     invariant_contract: &InvariantContract,
     call_result: RawCallResult,
     executor: &Executor,
     calldata: &[BasicTxDetails],
     failures: &mut InvariantFailures,
+    targeted_contracts: &FuzzRunIdentifiedContracts,
+    state_changeset: StateChangeset,
```

Comment:
why do we need them here?

looks expensive

Reply:
Right, to support handler assertion on invariants, we need to check if they've failed after execution is done, by checking DSTest's `failed()` func. So we need to check the targeted contracts which will have the handlers. And the fresh state is on the state changeset, so we also need it.

Reply:
Been also briefly testing this for any performance regression, doesn't seem to have any significant impact. On average a whole invariant campaign seems to take around 300ms-400ms more.